src/pset/finalizer.rs

```diff
@@ -204,6 +204,53 @@ pub(super) fn get_descriptor(
     }
 }
 
+// Helper function to de-duplicate code
+pub fn _interpreter_inp_check<C: secp256k1_zkp::Verification>(
+    pset: &Pset,
+    tx: &Transaction,
+    secp: &Secp256k1<C>,
+    index: usize,
+) -> Result<(), Error> {
+    let cltv = pset
+        .locktime()
+        .map_err(|_e| Error::LockTimeCombinationError)?;
+    let input = &pset.inputs[index];
+
+    let spk = get_scriptpubkey(pset, index).map_err(|e| Error::InputError(e, index))?;
+    let empty_script_sig = Script::new();
+    let empty_witness = Vec::new();
+    let script_sig = input.final_script_sig.as_ref().unwrap_or(&empty_script_sig);
```

Comment:
These `empty_script_sig` and `empty_witness` variables feel a bit weird. It's zero-cost to produce an empty `Script` or `Vec`, you can just do it inline.

Reply:
I think we need a reference to a Script or witness and I cannot do create Script inline and get a reference to it inline.